### ISSUE TITLE
 Failed assertion: line 22 pos 14: 'url != null': is not true.

### ISSUE THREAD

**Issue by daniele777** [sbis04/sign_in_flutter#12]
help i have this error ...

════════ Exception caught by widgets library ═══════════════════════════════════════════════════════
The following assertion was thrown building FirstScreen(dirty):
'package:flutter/src/painting/_network_image_io.dart': Failed assertion: line 22 pos 14: 'url != null': is not true.

**Comment by mayurvpatil** [comment]
facing same error

**Comment by mayurvpatil** [comment]
Fixed this issue after few try, 

Make sure you **provide SHA1 certificate** even if it's marked as optional on firebase console and **update your both build.gradle** files as instructed in step 3 ( Add firebase SDK ) or step 7 in article. This two thing fixed this issue for me.  

also 
 
### Make sure you follow [ Medium article ](https://medium.com/flutter-community/flutter-implementing-google-sign-in-71888bca24ed)  and don't miss a single step.

> I'm new to flutter but debugged this issue in detailed and it looks like google is not sending a proper  responce to your request (or null reponse) because so some missconfiguration, and  assert methods are just an validation to prevent null pointer kind of exception.

**Comment by mosesweb** [comment]
Also had this problem but got it working after cleaning and restarting Android Studio to reload the SHA1 google json file after I've entered it to the Firebase authentication. 

Next up I faced a similar problem that email was null too. Took me a while to understand that I've been using multiple providers for my account, and that the email was found on another element of the providerData List property of the current user.
